sys/include/net/gnrc/lwmac/lwmac.h

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *               2016 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_lwmac Simplest possible MAC layer
+ * @ingroup     net_gnrc
+ * @brief       Lightweight MAC protocol that allows for duty cycling to save
+ *              energy.
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the LWMAC protocol
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ */
+
+#ifndef GNRC_LWMAC_H
+#define GNRC_LWMAC_H
+
+#include <kernel_types.h>
+#include <net/gnrc/netdev2.h>
+
+/**
+ * @brief Forward gnrc_netdev2 declaration
+ */
+typedef struct gnrc_netdev2 gnrc_netdev2_t;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Time between consecutive wakeups. This parameter governs power consumption,
+ *        latency and throughput!
+ */
+#ifndef LWMAC_WAKEUP_INTERVAL_US
+#define LWMAC_WAKEUP_INTERVAL_US        (100U * 1000)
+#endif
+
+/**
+ * @brief The Maximum WR duration time.
+ */
+#ifndef LWMAC_PREAMBLE_DURATION_US
+#define LWMAC_PREAMBLE_DURATION_US      ((12*LWMAC_WAKEUP_INTERVAL_US)/10)
```

Comment:
To my knowledge, multiplying by `1.2` would result in an implicit cast to `float` which should be avoided. Integer division is enough.

```c
#define A 15
#define B ((A*12)/10)
#define C (1.2 * A)

printf("A = %i, B = %i, C = %i\n", A, B, C);
```

results in `A = 15, B = 18, C = 2147483640`